lib/Cron/Task.php

```diff
@@ -32,12 +109,116 @@ public function __construct() {
 	 * @return void
 	 */
 	protected function run($argument) {
-		if (!\OCP\App::isEnabled('files_antivirus')) {
+		if (!\OCP\App::isEnabled('files_antivirus')
+			|| $this->appConfig->getAvScanBackground() !== 'true'
+		) {
+			return;
+		}
+
+		// locate files that are not checked yet
+		try {
+			$result = $this->getFilesForScan();
+		} catch (\Exception $e) {
+			$this->logger->error(
+				__METHOD__ . ', exception: ' . $e->getMessage(),
+				['app' => 'files_antivirus']
+			);
 			return;
 		}
 
-		$application = new Application();
-		$container = $application->getContainer();
-		$container->query('BackgroundScanner')->run();
+		$cnt = 0;
+		while (($row = $result->fetch()) && $cnt < self::BATCH_SIZE) {
```

Comment:
Since the usual way is `while (($row = $result->fetch()) !== false)`, I guess that in this case it isn't possible the fetch a value that could be misinterpreted a false.